File: gramarogya_frontend/src/services/reportService.js
import axiosClient from "../api/axiosClient";

// =====================================================
// REPORT SUMMARY
// =====================================================

const fetchSummary = async () => {
  const response = await axiosClient.get("/reports/summary");
  return response.data;
};


// =====================================================
// BENEFICIARY REPORT
// =====================================================

const fetchBeneficiaryReport = async () => {
  const response = await axiosClient.get("/reports/beneficiaries");
  return response.data;
};


// =====================================================
// VISIT REPORT
// =====================================================

const fetchVisitReport = async () => {
  const response = await axiosClient.get("/reports/visits");
  return response.data;
};


// =====================================================
// INVENTORY REPORT
// =====================================================

const fetchInventoryReport = async () => {
  const response = await axiosClient.get("/reports/inventory");
  return response.data;
};


// =====================================================
// HEALTH RECORD REPORT
// =====================================================

const fetchHealthRecordReport = async () => {
  const response = await axiosClient.get("/reports/health-records");
  return response.data;
};


// =====================================================
// LOW STOCK REPORT
// =====================================================

const fetchLowStockReport = async () => {
  const response = await axiosClient.get(
    "/reports/inventory/low-stock"
  );

  return response.data;
};


// =====================================================
// OUT OF STOCK REPORT
// =====================================================

const fetchOutOfStockReport = async () => {
  const response = await axiosClient.get(
    "/reports/inventory/out-of-stock"
  );

  return response.data;
};


// =====================================================
// EXPORT
// =====================================================

const reportService = {
  fetchSummary,
  fetchBeneficiaryReport,
  fetchVisitReport,
  fetchInventoryReport,
  fetchHealthRecordReport,
  fetchLowStockReport,
  fetchOutOfStockReport,
};

export default reportService;

File: gramarogya-backend/src/main/java/com/gramarogya/gramarogya_backend/service/ReportService.java
package com.gramarogya.gramarogya_backend.service;

import com.gramarogya.gramarogya_backend.dto.BeneficiaryReportDto;
import com.gramarogya.gramarogya_backend.dto.Health_Records.HealthRecordReportDto;
import com.gramarogya.gramarogya_backend.dto.medicine.InventoryReportDto;
import com.gramarogya.gramarogya_backend.dto.report.ReportSummaryDto;
import com.gramarogya.gramarogya_backend.dto.visit.VisitReportDto;
import org.springframework.security.core.Authentication;

import java.util.List;

public interface ReportService {

    // =====================================================
    // DASHBOARD SUMMARY
    // =====================================================

    ReportSummaryDto getSummary(Authentication authentication);


    // =====================================================
    // BENEFICIARY REPORT
    // =====================================================

    List<BeneficiaryReportDto> getBeneficiaryReport(
            Authentication authentication
    );


    // =====================================================
    // VISIT REPORT
    // =====================================================

    List<VisitReportDto> getVisitReport(
            Authentication authentication
    );


    // =====================================================
    // INVENTORY REPORT
    // =====================================================

    List<InventoryReportDto> getInventoryReport(
            Authentication authentication
    );


    // =====================================================
    // HEALTH RECORD REPORT
    // =====================================================

    List<HealthRecordReportDto> getHealthRecordReport(
            Authentication authentication
    );


    // =====================================================
    // LOW STOCK
    // =====================================================

    List<InventoryReportDto> getLowStockReport(
            Authentication authentication
    );


    // =====================================================
    // OUT OF STOCK
    // =====================================================

    List<InventoryReportDto> getOutOfStockReport(
            Authentication authentication
    );
}